sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S2077_vb.html

```diff
@@ -0,0 +1,80 @@
+  <li> <code>System.Data.SqlClient.SqlDataAdapter.SqlDataAdapter(string, ...)</code> </li>
+  <li> <code>System.Data.Odbc.OdbcCommand.OdbcCommand(string, ...)</code> </li>
+  <li> <code>System.Data.Odbc.OdbcDataAdapter.OdbcDataAdapter(string, ...)</code> </li>
+  <li> <code>System.Data.SqlServerCe.SqlCeCommand.SqlCeCommand(string, ...)</code> </li>
+  <li> <code>System.Data.SqlServerCe.SqlCeDataAdapter.SqlCeDataAdapter(string, ...)</code> </li>
+  <li> <code>System.Data.OracleClient.OracleCommand.OracleCommand(string, ...)</code> </li>
+  <li> <code>System.Data.OracleClient.OracleDataAdapter.OracleDataAdapter(string, ...)</code> </li>
+  <li> <code>Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.ExecuteSqlCommand(...)</code> </li>
+  <li> <code>Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.ExecuteSqlCommandAsync(...)</code> </li>
+  <li> <code>Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSql&lt;TEntity&gt;(System.Linq.IQueryable&lt;TEntity&gt;,
+  System.FormattableString)</code> </li>
+  <li> <code>System.Data.SqlClient.SqlCommand.CommandText.set</code> </li>
+  <li> <code>System.Data.Odbc.OdbcCommand.CommandText.set</code> </li>
+  <li> <code>System.Data.SqlServerCe.SqlCeCommand.CommandText.set</code> </li>
+  <li> <code>System.Data.OracleClient.OracleCommand.CommandText.set</code> </li>
+</ul>
+<h2>Ask Yourself Whether</h2>
+<ul>
+  <li> the SQL query contains any non sanitized input from a user or another untrusted source. </li>
+</ul>
+<p>You are at risk if you answered yes to this question.</p>
+<h2>Recommended Secure Coding Practices</h2>
+<ul>
+  <li> Avoid building queries manually using concatenation or formatting. If you do it anyway, do not include user input in this building process.
+  </li>
+  <li> Use <a href="https://www.owasp.org/index.php/Query_Parameterization_Cheat_Sheet">parameterized queries, prepared statements, or stored
+  procedures</a> whenever possible. </li>
+  <li> You may also use libraries such as Hibernate which, if used correctly, reduce injection risks. </li>
+  <li> Avoid executing SQL queries containing unsafe input in stored procedures or functions. </li>
+  <li> <a href="https://www.owasp.org/index.php/Input_Validation_Cheat_Sheet">Sanitize</a> every unsafe input. </li>
+</ul>
+<p>You can also reduce the impact of an attack by using a database account with low privileges.</p>
+<h2>Questionable Code Example</h2>
+<pre>
+Public Sub SqlCommands(ByVal connection As SqlConnection, ByVal query As String)
+    Dim command As SqlCommand
+    command = New SqlCommand(query) ' Questionable
+
+    command.CommandText = query ' Questionable
+
+    Dim adapter As SqlDataAdapter
+    adapter = New SqlDataAdapter(query, connection) ' Questionable
+End Sub
+
+Public Sub Foo(ByVal context As DbContext, ByVal query As String)
+    context.Database.ExecuteSqlCommand(query) ' Questionable
+
+    context.Query(Of User)().FromSql(query) ' Questionable
+End Sub
```

Comment:
@valhristov I am adding "Exceptions" code examples showing that hard-coded strings and formatted strings will not raise an issue.